From 3c3f2115ed309157fc56c6d81c71d3142216490e Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Thu, 8 Apr 2021 20:01:38 +0300
Subject: [PATCH 277/302] staging: media: imx: isi: cap: Drop mbus_code
 parameter to mxc_isi_find_format()

The mxc_isi_find_format() function is always called with the mbus_code
parameter set to NULL. Drop it.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 drivers/staging/media/imx/imx8-isi-cap.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-isi-cap.c b/drivers/staging/media/imx/imx8-isi-cap.c
index 3aaf55152f07..b738c6af26e8 100644
--- a/drivers/staging/media/imx/imx8-isi-cap.c
+++ b/drivers/staging/media/imx/imx8-isi-cap.c
@@ -255,8 +255,7 @@ struct mxc_isi_fmt *mxc_isi_get_format(unsigned int index)
 /*
  * lookup mxc_isi color format by fourcc or media bus format
  */
-struct mxc_isi_fmt *mxc_isi_find_format(const u32 *pixelformat,
-					const u32 *mbus_code, int index)
+struct mxc_isi_fmt *mxc_isi_find_format(const u32 *pixelformat, int index)
 {
 	struct mxc_isi_fmt *fmt, *def_fmt = NULL;
 	unsigned int i;
@@ -269,8 +268,6 @@ struct mxc_isi_fmt *mxc_isi_find_format(const u32 *pixelformat,
 		fmt = &mxc_isi_out_formats[i];
 		if (pixelformat && fmt->fourcc == *pixelformat)
 			return fmt;
-		if (mbus_code && fmt->mbus_code == *mbus_code)
-			return fmt;
 		if (index == id)
 			def_fmt = fmt;
 		id++;
@@ -859,7 +856,7 @@ static int mxc_isi_cap_enum_fmt(struct file *file, void *priv,
 
 	dev_dbg(&isi_cap->pdev->dev, "%s\n", __func__);
 
-	fmt = mxc_isi_find_format(NULL, NULL, f->index);
+	fmt = mxc_isi_find_format(NULL, f->index);
 	if (!fmt)
 		return -EINVAL;
 
@@ -1215,7 +1212,7 @@ static int mxc_isi_cap_enum_framesizes(struct file *file, void *priv,
 	struct device_node *parent;
 	struct mxc_isi_fmt *fmt;
 
-	fmt = mxc_isi_find_format(&fsize->pixel_format, NULL, 0);
+	fmt = mxc_isi_find_format(&fsize->pixel_format, 0);
 	if (!fmt || fmt->fourcc != fsize->pixel_format)
 		return -EINVAL;
 
-- 
Regards,

Laurent Pinchart

